numpy/core/src/multiarray/ctors.c

```diff
@@ -3611,6 +3611,12 @@ PyArray_FromString(char *data, npy_intp slen, PyArray_Descr *dtype,
 
     binary = ((sep == NULL) || (strlen(sep) == 0));
     if (binary) {
+        if (DEPRECATE(
+                "The binary mode of fromstring is deprecated, and behaved"
```

Comment:
Needs space or newline at end here.

Reply:
Maybe "it" instead of "and".

Reply:
I think we should handle this in `array_fromstring`. In particular, the problem is in the `"s#"` format. We can check if the string argument is unicode and, if so, raise a `FutureWarning` that in the future only byte strings will be accepted. i.e., the user has to use an encoding to convert unicode to bytes. I suppose we could add an encoding keyword, but in this case I think only accepting byte strings should be OK. Note that `frombuffer` fails for unicode strings in Python 3, so we might also want to change that so that only byte strings are accepted.